pupepat/quiver.py

```diff
@@ -0,0 +1,65 @@
+import os
+from pupepat.surface import SurfaceFitter
+from pupepat.utils import get_inliers, offsets, estimate_scatter, prettify_focdmd
+from pupepat.plot import plot_quiver
+import numpy as np
+from astropy.io import ascii
+
+
+def make_quiver_plot(output_dir, output_table, output_plot='pupe-pat'):
+    """
+    Make a quiver plot for a full M2 focus grid
+    :param output_dir: Output Directory for the plot
+    :param output_table: Name of the ascii file with fit results
+    :param output_plot: Filename for the plot
+    """
+    # Read in the data table
+    data = ascii.read(os.path.join(output_dir, output_table))
+    data.sort('FOCDMD')
+
+    for demanded_focus in np.unique(data['FOCDMD']):
+
+        focus_set = data[data['FOCDMD'] == demanded_focus]
+        # Calculate outliers using plain centroid and center of the circles
+        centroid_inliers = get_inliers(offsets(focus_set['x0_centroid'], focus_set['y0_centroid'], focus_set['x0_outer'], focus_set['y0_outer']), 6.0)
+        focus_set = focus_set[centroid_inliers]
+
+        # Calculate outliers using nearby points
+        neighbor_inliers = get_neighbor_inliers(focus_set)
+        focus_set = focus_set[neighbor_inliers]
+
+        # Fit a smooth surface to inner and outer offsets
+        focus_surface = fit_focus_surface(focus_set)
+
+        # Make the actual plot
+        output_filename = os.path.join(output_dir, '{basename}-{focdmd}-quiver.pdf'.format(basename=output_plot,
+                                                                                      focdmd=prettify_focdmd(demanded_focus)))
+        plot_quiver(focus_set, focus_surface, output_filename)
+
+
+def fit_focus_surface(data):
+    fitter = SurfaceFitter()
+    dx_surface = fitter.fit(data['M2ROLL'], data['M2PITCH'], data['x0_outer'] - data['x0_inner'], 1)
```

Comment:
What's the 1 for?

Reply:
The one is a first order (linear) plane.

Reply:
Never mind, I got it

Reply:
Ah, I replied to my question and then refreshed the page and saw that you had answered!